query/src/main/java/org/apache/kylin/query/pushdown/PushDownRunnerOtherImpl.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kylin.query.pushdown;
+
+import org.apache.kylin.common.KylinConfig;
+import org.apache.kylin.common.util.DBUtils;
+import org.apache.kylin.metadata.querymeta.SelectedColumnMeta;
+import org.apache.kylin.sdk.datasource.framework.JdbcConnector;
+import org.apache.kylin.sdk.datasource.framework.SourceConnectorFactory;
+import org.apache.kylin.source.adhocquery.AbstractPushdownRunner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.LinkedList;
+import java.util.List;
+
+public class PushDownRunnerOtherImpl extends AbstractPushdownRunner {
```

Comment:
The name of `PushDownRunnerOtherImpl` is not well defined. Since it is the the advanced version of elder `PushDownRunnerJdbcImpl`, and reuse some logic in Data Source SDK. Maybe we change is name to something like `PushDownRunnerJdbcAdvencedImpl`? What do you think @shaofengshi ?